src/shortcut_formatting.ts

```diff
@@ -0,0 +1,95 @@
+import {ShortcutRegistry} from 'blockly';
+import {keyNames} from './keynames';
+
+const isMacPlatform = navigator.platform.startsWith('Mac');
+
+/**
+ * Format the primary shortcut for this platform in a user facing format.
+ *
+ * @param action The action name, e.g. "cut".
+ * @param format The key format.
+ * @returns The formatted shortcut.
+ */
+export function formatActionShortcut(
+  action: string,
+  format: ShortcutFormat,
+): string {
+  const parts = actionShortcutsForPlatform(action, format)[0];
+  return parts.join(isMacPlatform ? ' ' : ' + ');
+}
+
+const modifierNamesByFormat: Record<ShortcutFormat, Record<string, string>> = {
+  long: {
+    'Control': 'Ctrl',
+    'Meta': '⌘ Command',
+    'Alt': isMacPlatform ? '⌥ Option' : 'Alt',
+  },
+  short: {
+    'Control': 'Ctrl',
+    'Meta': '⌘',
+    'Alt': isMacPlatform ? '⌥' : 'Alt',
+  },
+};
+
+export type ShortcutFormat = 'long' | 'short';
+
+/**
+ * Find the relevant shortcuts for the given action for the current platform.
+ * Keys are returned in a user facing format.
+ *
+ * This could be considerably simpler if we only bound shortcuts relevant to the
+ * current platform or tagged them with a platform.
+ *
+ * @param action The action name, e.g. "cut".
+ * @param format The key format.
+ * @returns The formatted shortcuts.
+ */
+export function actionShortcutsForPlatform(
```

Comment:
generally I'd prefer having a `getShortActionShortcut` and `getLongActionShortcut` instead of making the user pass a magic string, if there's only two options.

Reply:
or potentially having one method with a boolean to switch to short mode or something like that

Reply:
Slightly complicated by there being short vs long but also individual keys vs joined string. Gone with API that looks nice for the callers in 6f692d428ee3400dd56c498a6db0ae876b5ceeb6 as in fact we only use short + joined string (for context menu) and long + individual keys (for help dialog).